src/Features/LanguageServer/Protocol/Handler/RoslynRequestExecutionQueue.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
+using System.Collections.Immutable;
+using CommonLanguageServerProtocol.Framework;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    internal class RoslynRequestExecutionQueue : RequestExecutionQueue<RequestContext>
+    {
+        private ImmutableArray<string> _supportedLanguages;
+
+        public RoslynRequestExecutionQueue(ImmutableArray<string> supportedLanguages, string serverKind, ILspServices services, ILspLogger logger)
+            : base(serverKind, services, logger)
+        {
+            _supportedLanguages = supportedLanguages;
+        }
+
+        public override Task<RequestContext?> CreateRequestContextAsync(IQueueItem<RequestContext> queueItem, CancellationToken cancellationToken)
```

Comment:
I wonder if this could move to IQueueItem, and if so, whether IQueueItem could become non-generic